salt/utils/gitfs.py

```diff
@@ -3045,6 +3045,61 @@ def checkout(self):
                 else:
                     self.pillar_dirs[cachedir] = env
 
+    @contextlib.contextmanager
+    def global_lock(self, timeout=0, poll_interval=0.5):
+        '''
+        Set and automatically clear a lock on all the repos
+        '''
+        if not isinstance(lock_type, six.string_types):
```

Comment:
`lock_type` isn't actually defined here - did you mean to take it as a parameter in the function?